src/richText.js

```diff
@@ -588,10 +570,85 @@ define([
         };
     }
 
+    function createFormModelForEditor(form) {
```

Comment:
Why not add a `getIconFromPath` method to the real form object and eliminate this adaptor? If you think `getIconFromPath` is not useful in other contexts, then I would recommend making it a private utility function in this module with the interface `getIconFromPath(path, form)`.